simulators/mujoco_simulator/revolve2/simulators/mujoco_simulator/_custom_mujoco_viewer.py

```diff
@@ -220,6 +222,17 @@ def _key_callback(
                 case _:
                     pass
 
+    def current_viewport_size(self) -> Tuple[int, int]:
```

Comment:
why is it updating and returning at the same time? Do all actions need both?

Reply:
Most of the actions should need neither. 
I put that in for the specific case of video rendering where the viewport size is fetched too earlier and results in an empty video file.
I find that the additional affectations are not too excessive compare to having outdated cached values. What is you take on that?